code/datums/actions/mobs/assume_form.dm

```diff
@@ -0,0 +1,86 @@
+/// Allows a mob to assume the form of another item or mob.
+/// Warning, this will likely shit the bricks if you add this action to anything more sophisticated than a basic mob- this isn't built for anything carbon-wise.
+/datum/action/cooldown/mob_cooldown/assume_form
+	name = "Assume Form"
+	desc = "Choose something that you wish to blend into the environment as. Click on yourself to reset your appearance."
+	button_icon_state = "sniper_zoom"
+	background_icon_state = "bg_alien"
+	overlay_icon_state = "bg_alien_border"
+	ranged_mousepointer = 'icons/effects/mouse_pointers/supplypod_target.dmi'
+	check_flags = AB_CHECK_CONSCIOUS
+	cooldown_time = 1.5 SECONDS
+
+	/// Stuff that we can not disguise as.
+	var/static/list/blacklist_typecache = typecacheof(list(
+		/atom/movable/screen,
+		/obj/effect,
+		/obj/energy_ball,
+		/obj/narsie,
+		/obj/singularity,
+	))
+
+/datum/action/cooldown/mob_cooldown/assume_form/Grant(mob/grant_to)
+	. = ..()
+	RegisterSignal(owner, COMSIG_LIVING_DEATH, PROC_REF(reset_appearances))
+
+/datum/action/cooldown/mob_cooldown/assume_form/Remove(mob/remove_from)
+	UnregisterSignal(owner, COMSIG_LIVING_DEATH)
```

Comment:
We don't reset appearances here. Are we ok with people getting stuck in an assumed form if the ability is removed somehow?

Reply:
yeah i figure i'll invoke it